pkg/openstack/loadbalancer_test.go

```diff
@@ -593,3 +595,296 @@ func Test_getIntFromServiceAnnotation(t *testing.T) {
+			name: "test get subnet from service annotation",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID:           "test-id",
+									FloatingNetworkID:  "test-floating-network-id",
+									FloatingSubnetID:   "test-floating-subnet-id",
+									FloatingSubnet:     "test-floating-subnet",
+									NetworkID:          "test-network-id",
+									FloatingSubnetTags: "test-floating-subnet-tags",
+									MemberSubnetID:     "test-member-subnet-id",
+								},
+							},
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/subnet-id": "test-id",
+							"loadbalancer.openstack.org/class":     "test-class",
+						},
+					},
+				},
+			},
+			want: "test-id",
+		},
+		{
+			name: "test get subnet from config class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID:           "test-class-subnet-id",
+									FloatingNetworkID:  "test-floating-network-id",
+									FloatingSubnetID:   "test-floating-subnet-id",
+									FloatingSubnet:     "test-floating-subnet",
+									NetworkID:          "test-network-id",
+									FloatingSubnetTags: "test-floating-subnet-tags",
+									MemberSubnetID:     "test-member-subnet-id",
+								},
+							},
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/class": "test-class",
+						},
+					},
+				},
+			},
+			want:        "test-class-subnet-id",
+			expectedErr: nil,
+		},
+		{
+			name: "test get subnet from config class with invalid loadbalancer class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							SubnetID: "test-subnet-id",
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/class": "test-class",
+						},
+					},
+				},
+			},
+			want:        "",
+			expectedErr: fmt.Errorf("invalid loadbalancer class %q", "test-class"),
+		},
+		{
+			name: "test get subnet from default config",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							SubnetID: "test-default-subnet-id",
+						},
+					},
+				},
+				service: &corev1.Service{},
+			},
+			want:        "test-default-subnet-id",
+			expectedErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.args.lbaasV2.getSubnetID(tt.args.service, tt.args.svcConf)
+			if tt.expectedErr != nil {
```

Comment:
Watch out! You are not checking the case where `tt.expectedErr` is nil and `err` is not nil

Reply:
yea!! that's what I'm referring to as well, checking it is repetitive since the assert checks it, as shown in this screenshot
<img width="750" alt="Screenshot 2023-10-10 at 13 17 47" src="https://github.com/kubernetes/cloud-provider-openstack/assets/69452689/aae57cff-fef8-47f7-af84-f6b0b668b28a">

Reply:
Am I blind? I think your code only checks err if tt.expectedErr is not nil.